ddtrace/internal/globaltracer.go

```diff
@@ -0,0 +1,43 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2025 Datadog, Inc.
+
+package internal
+
+import "sync/atomic"
+
+var (
+	// globalTracer stores the current tracer as *ddtrace/tracer.Tracer (pointer to interface). The
+	// atomic.Value type requires types to be consistent, which requires using the same type for the
+	// stored value.
+	globalTracer atomic.Value
+)
+
+// tracerLike is an interface to restrict the types that can be stored in `globalTracer`.
+// This interface doesn't leak to the users. We are leveraging the type system to generating
+// the functions below for `tracer.Tracer` without creating an import cycle.
+type tracerLike interface {
+	Flush()
+	Stop()
+}
+
+// StoreGlobalTracer stores a tracer in the global tracer.
+// It is the responsibility of the caller to ensure that the value is `tracer.Tracer`.
+func StoreGlobalTracer[T tracerLike](t T) {
```

Comment:
Do we actually need an additional StoreGlobalTracer method? Can't we just use the SetGlobalTracer? What's the additional benefit?

Reply:
This replaces the original `Store` operation at the `init` function. I agree it could be replaced by the `SetGlobalTracer`, but I'm not sure why it was decided to go this way.